node_modules/oae-principals/lib/test/util.js

```diff
@@ -1047,3 +1125,256 @@ var assertUpdateUserFails = module.exports.assertUpdateUserFails = function(rest
+            assert.ok(!err);
+
+            // Verify the token is gone
+            return assertDeleteEmailTokenFails(restCtx, userId, 404, callback);
+        });
+    });
+};
+
+/**
+ * Assert that an email token can be deleted
+ *
+ * @param  {RestContext}    restCtx         The REST context of a user who will perform the request
+ * @param  {String}         userId          The id of the user for which to delete the pending email token
+ * @param  {Number}         httpCode        The expected HTTP code of the failed has pending email token operation
+ * @param  {Function}       callback        Invoked when the delete operation has been performed
+ * @throws {AssertionError}                 Thrown if the operation is successful
+ */
+var assertDeleteEmailTokenFails = module.exports.assertDeleteEmailTokenFails = function(restCtx, userId, httpCode, callback) {
+    RestAPI.User.deleteEmailToken(restCtx, userId, function(err, data) {
+        assert.ok(err);
+        assert.strictEqual(err.code, httpCode);
+        return callback();
+    });
+};
+
+/**
+ * Assert that the correct user ids are mapped to an email address
+ *
+ * @param  {String}     email               The email address for which to check the mapping
+ * @param  {String[]}   expectedUserIds     The expected ids of the users who should be mapped to the email address
+ * @param  {Function}   callback            Invoked when the mapping has been verified
+ * @throws {AssertionError}                 Thrown if the operation is unsuccessful or the mapping is incorrect
+ */
+var assertUserEmailMappingEquals = module.exports.assertUserEmailMappingEquals = function(email, expectedUserIds, callback) {
+    PrincipalsDAO.getUsersByEmail(email.toLowerCase(), function(err, usersByEmail) {
+        assert.ok(!err);
+
+        // Get the ids of the mapped users, don't care about sorting
+        var userIds = _.chain(usersByEmail)
+            .pluck('id')
+            .sort()
+            .value();
+        expectedUserIds = expectedUserIds.sort();
+
+        // Assert the retrieved user ids are what we expected
+        assert.deepEqual(userIds, expectedUserIds);
+        return callback();
+    });
+};
+
+/**
+ * Execute a callback function once a verification email is sent
+ *
+ * @param  {String}     [email]             The email address too which the token should have been sent
+ * @param  {Boolean}    expectAdminMessage  Whether it's expected that the message was triggered by an admin
+ * @param  {Function}   callback            Invoked once the verification email was sent to the user
+ * @param  {String}     callback.token      The email verification token that was sent to the user
+ * @throws {AssertionError}                 Thrown if an email was sent to the wrong email address or does not contain a proper token
+ */
+var onceVerificationEmailSent = module.exports.onceVerificationEmailSent = function(email, expectAdminMessage, callback) {
```

Comment:
Can we make the `expectAdminMessage` an object instead with key `expectAdminMessage`?